lib/std/os.zig

```diff
@@ -3845,6 +3847,21 @@ pub fn pipe2(flags: u32) PipeError![2]fd_t {
         };
         return fds;
     }
+    if (comptime std.Target.current.isHaiku()) {
```

Comment:
Isn't this identical to the Darwin code path?

Reply:
yes, updated conditional and re-used Darwin code path instead